### PR TITLE
[OYPD-287] Data layer base configuration feature.

### PR DESCRIPTION
**Pull request by Kerby70**
Jira https://propeople-us.atlassian.net/browse/OYPD-287
D.o https://www.drupal.org/node/2857039

Steps to review:
- [x] View content pages for the following content types
  - Branch
  - Camp
  - Facility
  - Blog
  - Program
  - Program Subcategory
  - Membership
  - Landing pages
- [x] Open Chrome console
- [x] Enter dataLayer
- [x] Make sure you see fields

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>25</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build632/build_reports/Drupalsniff.txt
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build632 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build632 |
| Installation process | http://install.openy-dev.ffwua.com/build632/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/632/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build632/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build633 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build633 |
| Installation process | http://install.openy-dev.ffwua.com/build633/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/633/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build633/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build637 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build637 |
| Installation process | http://install.openy-dev.ffwua.com/build637/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/637/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build637/profiles/openy/build/reports/behat

**Comment by alexschedrov** [comment]
1. Please add upgrade path for this feature.
- Enable module
- Revert all config changes
Eventually this should be testable on Upgrade Path environment

2. Can we rewrite steps for review a little bit?
- Open followign content type pages
-- ....
-- ...
- Open Chrome console
- Enter `dataLayer`
- Make sure you see fields

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
<strong>4</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build645/build_reports/Drupalsniff.txt
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build645 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build645 |
| Installation process | http://install.openy-dev.ffwua.com/build645/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/645/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build645/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build646 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build646 |
| Installation process | http://install.openy-dev.ffwua.com/build646/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/646/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build646/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

File: openy.install [hunk 1]
```diff
@@ -60,3 +60,29 @@ function openy_update_8003() {
...
+    'openy_node_blog',
+    'openy_node_category',
+    'openy_node_mbrshp',
+    'openy_node_program',
+  ];
+  $module_handler = \Drupal::service('module_handler');
+  $modules = array_filter($modules, function ($module) use ($module_handler) {
+    return $module_handler->moduleExists($module);
+  });
+  \Drupal::service('features.manager')->import($modules);
```
Comment (alexschedrov) [inline openy.install]: We shouldn't revert whole feature. There are diffrerent levels that can be used in upgrade path:
- Revert feature
- Revert some specific file
- Revert some specific property in config file

In upgrade path for the distribution we should use only 3rd, https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md#revert-only-specific-property-from-config.

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build648 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build648 |
| Installation process | http://install.openy-dev.ffwua.com/build648/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/648/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build648/profiles/openy/build/reports/behat

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build657 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build657 |
| Installation process | http://install.openy-dev.ffwua.com/build657/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/657/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build657/profiles/openy/build/reports/behat

**Comment by Kerby70** [comment]
Seeing an error in `Scenario: Create basic program and subcategory and check fields`

In logged errors on a new program subcategory page:
Error: Call to a member function getString() on null in _datalayer_field_get_value() (line 567 of /var/www/build657/profiles/openy/modules/contrib/datalayer/datalayer.module) #0 /var/www/build657/profiles/openy/modules/contrib/datalayer/datalayer.module(534): _datalayer_field_get_value(Object(Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem), 'entity_referenc...') #1 /var/www/build657/profiles/openy/modules/contrib/datalayer/datalayer.module(374): _datalayer_get_field_values(Object(Drupal\node\Entity\Node)) #2 /var/www/build657/profiles/openy/modules/contrib/datalayer/datalayer.module(212): _datalayer_get_entity_data(Object(Drupal\node\Entity\Node)) #3 /var/www/build657/profiles/openy/modules/contrib/datalayer/datalayer.module(93): datalayer_get_page_data() #4 /var/www/build657/profiles/openy/modules/contrib/datalayer/datalayer.module(124): datalayer_get_data_from_page() #5 /var/www/build657/core/lib/Drupal/Core/Render/MainContent/HtmlRenderer.php(297): datalayer_page_attachments(Array) #6 /var/www/build657/core/lib/Drupal/Core/Render/MainContent/HtmlRenderer.php(273): Drupal\Core\Render\MainContent\HtmlRenderer->invokePageAttachmentHooks(Array) #7 /var/www/build657/core/lib/Drupal/Core/Render/MainContent/HtmlRenderer.php(117): Drupal\Core\Render\MainContent\HtmlRenderer->prepare(Array, Object(Symfony\Component\HttpFoundation\Request), Object(Drupal\Core\Routing\CurrentRouteMatch)) #8 /var/www/build657/core/lib/Drupal/Core/EventSubscriber/MainContentViewSubscriber.php(90): Drupal\Core\Render\MainContent\HtmlRenderer->renderResponse(Array, Object(Symfony\Component\HttpFoundation\Request), Object(Drupal\Core\Routing\CurrentRouteMatch)) #9 /var/www/build657/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php(111): Drupal\Core\EventSubscriber\MainContentViewSubscriber->onViewRenderArray(Object(Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent), 'kernel.view', Object(Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher)) #10 /var/www/build657/vendor/symfony/http-kernel/HttpKernel.php(149): Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher->dispatch('kernel.view', Object(Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent)) #11 /var/www/build657/vendor/symfony/http-kernel/HttpKernel.php(64): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1) #12 /var/www/build657/core/lib/Drupal/Core/StackMiddleware/Session.php(57): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #13 /var/www/build657/core/lib/Drupal/Core/StackMiddleware/KernelPreHandle.php(47): Drupal\Core\StackMiddleware\Session->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #14 /var/www/build657/core/modules/page_cache/src/StackMiddleware/PageCache.php(99): Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #15 /var/www/build657/core/modules/page_cache/src/StackMiddleware/PageCache.php(78): Drupal\page_cache\StackMiddleware\PageCache->pass(Object(Symfony\Component\HttpFoundation\Request), 1, true) #16 /var/www/build657/core/lib/Drupal/Core/StackMiddleware/ReverseProxyMiddleware.php(47): Drupal\page_cache\StackMiddleware\PageCache->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #17 /var/www/build657/core/lib/Drupal/Core/StackMiddleware/NegotiationMiddleware.php(50): Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #18 /var/www/build657/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #19 /var/www/build657/core/lib/Drupal/Core/DrupalKernel.php(652): Stack\StackedHttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true) #20 /var/www/build657/index.php(19): Drupal\Core\DrupalKernel->handle(Object(Symfony\Component\HttpFoundation\Request)) #21 {main}.
`

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build659 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build659 |
| Installation process | http://install.openy-dev.ffwua.com/build659/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/659/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build659/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============

--------------
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/662/

**Build Log**
*last 25 lines*

 ```
[...truncated 222 lines...]
stdout: You are about to create a /var/www/build662/sites/default/settings.php file and DROP all tables in your 'drupal662' database. Do you want to continue? (y/n): y

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/var/lib/jenkins/reinstall.retry

localhost                  : ok=14   changed=6    unreachable=0    failed=1   

FATAL: command execution failed
hudson.AbortException: Ansible playbook execution failed
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:227)
	at org.jenkinsci.plugins.ansible.AnsiblePlaybookBuilder.perform(AnsiblePlaybookBuilder.java:200)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:75)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:537)
	at hudson.model.Run.execute(Run.java:1741)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:381)
ERROR: Ansible playbook execution failed
```

**Comment by Sanchiz** [comment]
retest this please

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation | http://openy-dev.ffwua.com/build663 |
| Upgraded(upgrade path) installation | http://upgrade.openy-dev.ffwua.com/build663 |
| Installation process | http://install.openy-dev.ffwua.com/build663/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/663/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build663/profiles/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]
Approved.

**Pull request merged**